docs/architecture/raft.rst

```diff
@@ -44,7 +44,12 @@ Enabling Raft
 
     See :doc:`the upgrade guide from 5.2 to 5.4 </upgrade/index>` for details.
 
-ScyllaDB Open Source 5.2 and later, and ScyllaDB Enterprise 2023.1 and later come equipped with a procedure that can setup Raft-based consistent cluster management in an existing cluster. We refer to this as the **Raft upgrade procedure** (do not confuse with the :doc:`ScyllaDB version upgrade procedure </upgrade/index/>`).
+.. warning::
+    Raft is mandatory in ScyllaDB Open Source 6.0 and later, and ScyllaDB Enterprise 2024.2 and later. You cannot prevent Raft from getting enabled upon upgrade to 6.0 or 2024.2.
+
+.. TODO: add reference to the upgrade guide from 5.4 to 6.0
```

Comment:
When we add this new upgrade doc, we'll need to include a big note saying "Raft shall be enforced upon you with no way out. Prepare for quorum requirement" or sth (cc @kostja @gleb-cloudius )

Reply:
Hmm... @annastuchlik @tzach what do you think of the following idea:

Let's have a "template" upgrade guide present at all times in master branch: "Upgrade guide from <last-release> to NEW_RELEASE_PLACEHOLDER". It would be hidden from users, but visible to developers, so we can update it as we create new PRs. We can also refer to it in other docs; the paragraphs or sentences that refer to it would be hidden. Or maybe NEW_RELEASE_PLACEHOLDER would be automatically replaced with "upcoming release" or something that makes sense.

Once new release is branched, we would have an automatic process that copies this template guide and replaces NEW_RELEASE_PLACEHOLDER with the actual version.

Reply:
Just to clarify, from the previous version to *this* version, where *this* version is the one being documented. So 6.0 documentation should explain how to document to 6.0.